tests/background/chrome-api-test.js

```diff
@@ -87,4 +91,32 @@ describe('getChromeAPI', () => {
 
     assert.equal(error, fakeChrome.runtime.lastError);
   });
+
+  describe('APIs that require optional permissions', () => {
+    it('rejects if permission has not been granted', async () => {
+      const chromeAPI = getChromeAPI(fakeChrome);
+
+      let error;
+      try {
+        await chromeAPI.webNavigation.getAllFrames();
```

Comment:
This feels a little magical. Is it that the getter for `webNavigation` is not mocked, so it will behave in its "real" way here, which is to deny? Can you add some commenting?

Reply:
Setting the `fakeChrome.webNavigation` property simulates the permission having been granted. I'll add a comment.